newton/_src/geometry/types.py

```
@@ -1026,3 +1029,364 @@ def __hash__(self) -> int:
+
+        if method == "convex_hull":
+            from .utils import remesh_convex_hull  # noqa: PLC0415
+
+            hull_verts, hull_faces = remesh_convex_hull(self._positions)
+            return Mesh(hull_verts, hull_faces, compute_inertia=True)
+        elif method == "alphashape":
+            from .utils import remesh_alphashape  # noqa: PLC0415
+
+            hull_verts, hull_faces = remesh_alphashape(self._positions)
+            return Mesh(hull_verts, hull_faces, compute_inertia=True)
+        raise ValueError(f"Unsupported proxy mesh method: {method!r}. Supported: 'convex_hull'.")
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)